internal/jobs/launch_instance_gcp.go

```diff
@@ -0,0 +1,122 @@
+
+	// Detail information
+	Detail *models.GCPDetail `json:"detail"`
+
+	// GCP image name as fetched from image builder
+	ImageName string `json:"image_name"`
+
+	// The project id from Sources which is linked to a specific source
+	ProjectID string `json:"project_id"`
+}
+
+// Unmarshall arguments and handle error
+func HandleLaunchInstanceGCP(ctx context.Context, job dejq.Job) error {
+	args := LaunchInstanceGCPTaskArgs{}
+	err := decodeJob(ctx, job, &args)
+	if err != nil {
+		return err
+	}
+
+	ctx = contextLogger(ctx, job.Type(), args, args.AccountID, args.ReservationID)
+
+	jobErr := handleLaunchInstanceGCP(ctx, &args)
+
+	finishJob(ctx, args.ReservationID, jobErr)
+	return jobErr
+}
+
+// Job logic, when error is returned the job status is updated accordingly
+func handleLaunchInstanceGCP(ctx context.Context, args *LaunchInstanceGCPTaskArgs) error {
+	ctxLogger := ctxval.Logger(ctx)
+	ctxLogger.Debug().Msg("Started launch instance GCP job")
+
+	ctx = ctxval.WithAccountId(ctx, args.AccountID)
+	logger := ctxLogger.With().Int64("reservation", args.ReservationID).Logger()
+	logger.Info().Interface("args", args).Msg("Processing launch instance GCP job")
+
+	// status updates before and after the code logic
+	updateStatusBefore(ctx, args.ReservationID, "Launching instance(s)")
+	defer updateStatusAfter(ctx, args.ReservationID, "Launched instance(s)", 1)
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot create instances REST client: %w", err)
+	}
+	defer instancesClient.Close()
+
+	req2 := &computepb.BulkInsertInstanceRequest{
+		Project: args.ProjectID,
+		Zone:    args.Zone,
+		BulkInsertInstanceResourceResource: &computepb.BulkInsertInstanceResource{
+			NamePattern: ptr.String("inst-####"),
+			Count:       ptr.Int64(args.Detail.Amount),
+			MinCount:    ptr.Int64(args.Detail.Amount),
+			InstanceProperties: &computepb.InstanceProperties{
+				Disks: []*computepb.AttachedDisk{
+					{
+						InitializeParams: &computepb.AttachedDiskInitializeParams{
+							DiskSizeGb:  ptr.Int64(30),
+							SourceImage: ptr.String(args.ImageName),
+						},
+						AutoDelete: ptr.Bool(true),
+						Boot:       ptr.Bool(true),
+						Type:       ptr.String(computepb.AttachedDisk_PERSISTENT.String()),
+					},
+				},
+				MachineType: ptr.String(args.Detail.MachineType),
+				NetworkInterfaces: []*computepb.NetworkInterface{
+					{
+						Name: ptr.String("global/networks/default"),
+					},
+				},
+				Metadata: &computepb.Metadata{
+					Items: []*computepb.Items{
+						{
+							Key:   ptr.String("ssh-keys"),
+							Value: ptr.String(args.PubkeyID),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	logger.Trace().Msg("Executing Insert")
+	op, err := instancesClient.BulkInsert(ctx, req2)
+	if err != nil {
+		return fmt.Errorf("cannot bulk insert instances: %w", err)
+	}
+
+	// TODO: info regarding the reservation and info about the created instances (create a db record for each instance)
```

Comment:
Honestly, I would rather see this feature in the PR since you also add the field into the API (payload,  model). Or you can keep it this way and comment it out from there.